### PR TITLE
feat(log): check tty by atty for the log color or not

### LINKED ISSUE #5738: Slim down databend log
**Summary**

Maybe remove log file color format string. 

$more query.log
>"query.log" may be a binary file.  See it anyway?

```
ESC[2m2022-05-31T06:45:32.490093ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0mESC[1mConnectionESC[0mESC[1m{ESC[0mESC[3mpeerESC[0mESESC[2m=ESC[0mClientESC[1m}ESC[0mESC[2m:ESC[0m ESC[2mh2::codec::framed_writeESC[0mESC[2m:ESC[0m send ESC[3mframeESC[0mESC[2m=ESC[0mHeaders { stream_id: StreamId(3), flags: (0x4: END_HEADERS) }
ESC[2m2022-05-31T06:45:32.490128ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0mESC[1mConnectionESC[0mESC[1m{ESC[0mESC[3mpeerESC[0mESESC[2m=ESC[0mClientESC[1m}ESC[0mESC[2m:ESC[0m ESC[2mh2::codec::framed_writeESC[0mESC[2m:ESC[0m send ESC[3mframeESC[0mESC[2m=ESC[0mData { stream_id: StreamId(3) }
ESC[2m2022-05-31T06:45:32.490146ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0mESC[1mConnectionESC[0mESC[1m{ESC[0mESC[3mpeerESC[0mESESC[2m=ESC[0mClientESC[1m}ESC[0mESC[2m:ESC[0m ESC[2mh2::codec::framed_writeESC[0mESC[2m:ESC[0m send ESC[3mframeESC[0mESC[2m=ESC[0mData { stream_id: StreamId(3), flags: (0x1: END_STREAM) }
ESC[2m2022-05-31T06:45:32.490974ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0mESC[1mConnectionESC[0mESC[1m{ESC[0mESC[3mpeerESC[0mESESC[2m=ESC[0mClientESC[1m}ESC[0mESC[2m:ESC[0m ESC[2mh2::codec::framed_readESC[0mESC[2m:ESC[0m received ESC[3mframeESC[0mESC[2m=ESC[0mHeaders { stream_id: StreamId(3), flags: (0x4: END_HEADERS) }
ESC[2m2022-05-31T06:45:32.491010ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0mESC[1mConnectionESC[0mESC[1m{ESC[0mESC[3mpeerESC[0mESESC[2m=ESC[0mClientESC[1m}ESC[0mESC[2m:ESC[0m ESC[2mh2::codec::framed_readESC[0mESC[2m:ESC[0m received ESC[3mframeESC[0mESC[2m=ESC[0mData { stream_id: StreamId(3) }
ESC[2m2022-05-31T06:45:32.491026ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0mESC[1mConnectionESC[0mESC[1m{ESC[0mESC[3mpeerESC[0mESESC[2m=ESC[0mClientESC[1m}ESC[0mESC[2m:ESC[0m ESC[2mh2::codec::framed_readESC[0mESC[2m:ESC[0m received ESC[3mframeESC[0mESC[2m=ESC[0mHeaders { stream_id: StreamId(3), flags: (0x5: END_HEADERS | END_STREAM) }
ESC[2m2022-05-31T06:45:32.491099ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0m ESC[2mcommon_meta_grpc::grpc_clientESC[0mESC[2m:ESC[0m MetaGrpcClient::do_read reply ESC[3mreplyESC[0mESC[2m=ESC[0mOk(Response { metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Tue, 31 May 2022 06:45:32 GMT", "grpc-status": "0"} }, message: RaftReply { data: "null", error: "" }, extensions: Extensions })
ESC[2m2022-05-31T06:45:32.491153ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0m ESC[2mcommon_meta_grpc::grpc_clientESC[0mESC[2m:ESC[0m MetaGrpcClient send response to the handle ESC[3mrespESC[0mESC[2m=ESC[0mOk(Get(None))
ESC[2m2022-05-31T06:45:32.491411ZESC[0m ESC[34mDEBUGESC[0m ESC[2mcommon_meta_api::schema_api_implESC[0mESC[2m:ESC[0m get_database ESC[3mdb_id_seqESC[0mESC[2m=ESC[0m0 ESC[3mdb_idESC[0mESC[2m=ESC[0m0 ESC[3mname_keyESC[0mESC[2m=ESC[0mDatabaseNameIdent { tenant: "tenant1", db_name: "default" }
ESC[2m2022-05-31T06:45:32.491460ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0m ESC[2mcommon_meta_grpc::grpc_clientESC[0mESC[2m:ESC[0m MetaGrpcClient recv request ESC[3mreqESC[0mESC[2m=ESC[0mClientWorkerRequest { resp_tx: Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: true, is_tx_task_set: false } }) }, req: Get(GetKVReq { key: "__fd_db_id_list/tenant1/default" }) }
ESC[2m2022-05-31T06:45:32.491507ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0m ESC[2mcommon_meta_grpc::grpc_clientESC[0mESC[2m:ESC[0m MetaGrpcClient::do_read request EESC[3mreqESC[0mESC[2m=ESC[0mGetKV(GetKVReq { key: "__fd_db_id_list/tenant1/default" })
ESC[2m2022-05-31T06:45:32.491539ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0m ESC[2mcommon_meta_grpc::grpc_clientESC[0mESC[2m:ESC[0m MetaGrpcClient::do_read serialized request ESC[3mreqESC[0mESC[2m=ESC[0mRequest { metadata: MetadataMap { headers: {} }, message: RaftRequest { data: "{\"GetKV\":{\"key\":\"__fd_db_id_list/tenant1/default\"}}" }, extensions: Extensions }
ESC[2m2022-05-31T06:45:32.491595ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0m ESC[2mcommon_base::containers::poolESC[0mESC[2m:ESC[0m build new item of key: "10.16.56.47:28302"
ESC[2m2022-05-31T06:45:32.491636ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0m ESC[2mhyper::client::connect::httpESC[0mESC[2m:ESC[0m connecting to 10.16.56.47:28302
ESC[2m2022-05-31T06:45:32.491889ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0m ESC[2mhyper::client::connect::httpESC[0mESC[2m:ESC[0m connected to 10.16.56.47:28302
ESC[2m2022-05-31T06:45:32.491914ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0m ESC[2mh2::clientESC[0mESC[2m:ESC[0m binding client connection
ESC[2m2022-05-31T06:45:32.491939ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0m ESC[2mh2::clientESC[0mESC[2m:ESC[0m client connection bound
ESC[2m2022-05-31T06:45:32.491961ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0m ESC[2mh2::codec::framed_writeESC[0mESC[2m:ESC[0m send ESC[3mframeESC[0mESC[2m=ESC[0mSettings { flags: (0x0), enable_push: 0, initial_window_size: 2097152, max_frame_size: 16384 }
ESC[2m2022-05-31T06:45:32.492022ZESC[0m ESC[34mDEBUGESC[0m ESC[1mworker_loopESC[0mESC[2m:ESC[0mESC[1mdo_readESC[0mESC[2m:ESC[0mESC[1mmake_clientESC[0mESC[2m:ESC[0m ESC[2mcommon_base::containers::poolESC[0mESC[2m:ESC[0m build new item of key res: Ok(Channel)
:
```

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* ./target/debug/databend-query:
image

* ./target/debug/databend-query > query.log 2>&1&
image

## Changelog

- Improvement
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #5738

### CODE REVIEW

File: common/tracing/src/logging.rs [hunk 1]
```diff
@@ -74,12 +73,8 @@ pub fn init_global_tracing(app_name: &str, dir: &str, level: &str) -> Vec<Worker
     // Enable log compatible layer to convert log record to tracing span.
     LogTracer::init().expect("log tracer must be valid");
 
-    // Stdout layer.
-    let (stdout_writer, stdout_guard) = tracing_appender::non_blocking(std::io::stdout());
```
Comment (BohuTANG) [inline common/tracing/src/logging.rs:78]: The default layer is stdout, seems no need set it here, if we set it, the stdout will be duplicated.

Comment (drmingdrmer) [reply common/tracing/src/logging.rs:78]: query_logger is only enabled by session_mgr, by removing the stdout logger, will there be some log missing?:
https://github.com/datafuselabs/databend/blob/0fe675e92b3d8cd1d1aefab46b1729a08dbf0aa6/query/src/sessions/session_mgr.rs#L103-L108

Comment (drmingdrmer) [reply common/tracing/src/logging.rs:78]: Sorry I was speaking about another thing...

But what is the default layer?

Comment (BohuTANG) [reply common/tracing/src/logging.rs:78]: From my test, it's stdout

Comment (BohuTANG) [reply common/tracing/src/logging.rs:78]: The query log works well:
image

**Review by everpcpc — APPROVED** [review]

File: common/tracing/src/logging.rs [hunk 2]
```diff
@@ -94,16 +89,17 @@ pub fn init_global_tracing(app_name: &str, dir: &str, level: &str) -> Vec<Worker
     let jaeger_layer = Some(tracing_opentelemetry::layer().with_tracer(tracer));
 
     // Use env RUST_LOG to initialize log if present.
-    // Otherwise use the specified level.
+    // Otherwise, use the specified level.
     let directives = env::var(EnvFilter::DEFAULT_ENV).unwrap_or_else(|_x| level.to_string());
     let env_filter = EnvFilter::new(directives);
     let subscriber = Registry::default()
+        .with(fmt::layer().with_ansi(atty::is(atty::Stream::Stdout)))
```
Comment (Xuanwo) [inline common/tracing/src/logging.rs:96]: Should we output our log to `stderr`?

Comment (BohuTANG) [reply common/tracing/src/logging.rs:96]: I guess users usually use `2>&1` to do that?

**Comment by BohuTANG** [comment]
> I wonder why we need to redirect the stdout to file when there is already a `file_logging_layer`.

Hmm, `./databend-query -c ./databend-query.toml > query.log 2>&1 &` is for human.
file_logging_layer is JSON format is for the machine :)

**Review by drmingdrmer — APPROVED** [review]

**Comment by Xuanwo** [comment]
> I wonder why we need to redirect the stdout to file when there is already a `file_logging_layer`.

I'm second to this concern either.

Normally, most applications should only write logs once and let users decide how to store/use them.

As recommended by [12factor](https://12factor.net/), app should only write logs to `stdout`/`stderr`.

And there are also apps allowing users to do more configuring:

- log type: stdout/file/log drainer
- log format: text/json/colorful
- log path: /path/to/log
- even rotation and so on

**Pull request merged**